### PR TITLE
Add the Saboteur module to syndicate cyborgs

### PR DESCRIPTION
**Pull request by datlo**
This PR adds the syndicate saboteur ported and modified from tg as an option when selecting a syndicate cyborg's module. Built around more of a stealth role and capable of working solo, the Saboteur is a modified engineering cyborg capable of heavily crippling the station through controlled plasma fires, inconvenient barricades, breaches, assassinations of key targets, and more. 
saboteur1

Features : 

- Most tools and devices found on a standard engineering cyborg. The RCD is upgraded to consume less power when used, and like the engiborgs' can eat reinforced walls.

- A self repair upgrade

- A standard energy sword, a cryptographic sequencer, and an operative pinpointer (points to the nearest nuke op, not the disk)

- The ability to mail itself by disposals, like maintenance drones can.

- A cyborg chameleon projector. When turned on, after a delay, you'll look like a NT engi cyborg until you're damaged or punched. Consumes power when toggled, and to maintain.
saboteur3
saboteur4

- The ability to rename itself.

This PR also fixes an exploit that lets you refund borg spawners even after use by deleting the spawner once you actually get a borg.

🆑
Add: Nuclear operatives can now select the Saboteur module when picking a cyborg, a modified engineering module designed around stealth and sabotage.
Fix: Fixed an exploit allowing you to get free syndicate cyborgs.
/🆑

### CODE REVIEW

**Comment by datlo** [comment]
saboteur6

DISCOOO INFERNOOOO

**Comment by AffectedArc07** [comment]
image
hmmm

**Comment by KasparoVy** [comment]
Fuuuck yeeeees. Most likely never going to play this but the concept and sprites are awesome af

**Comment by datlo** [comment]
Travis plox

File: code/game/objects/items/devices/chameleonproj.dm [hunk 1]
```diff
@@ -148,3 +148,98 @@
...
+		for(i=1, i<=7, ++i)
+			do
+				X = 60*rand() - 30
+				Y = 60*rand() - 30
+				rsq = X*X + Y*Y
+			while(rsq<100 || rsq>900)
+			user.filters += filter(type="wave", x=X, y=Y, size=rand()*2.5+0.5, offset=rand())
+		for(i=1, i<=7, ++i)
+			f = user.filters[start+i]
+			animate(f, offset=f:offset, time=0, loop=3, flags=ANIMATION_PARALLEL)
```
[the remark this answers is not in the document]
Comment (datlo) [reply code/game/objects/items/devices/chameleonproj.dm]: this is code to animate the transition while the chameleon starts up
I'll admit I have no idea how it actually works

[the remark this answers is not in the document]
Comment (Citinited) [reply code/game/objects/items/devices/chameleonproj.dm]: Ahh okay

I'll admit I also have no idea what it does 😄

File: code/game/objects/items/devices/chameleonproj.dm [hunk 2]
```diff
@@ -148,3 +148,102 @@
...
+				X = 60 * rand() - 30
+				Y = 60 * rand() - 30
+				rsq = X * X + Y * Y
+			while(rsq < 100 || rsq > 900)
+			user.filters += filter(type = "wave", x = X, y = Y, size = rand() * 2.5 + 0.5, offset = rand())
+		for(i in 1 to 7)
+			f = user.filters[start+i]
+			animate(f, offset = f:offset, time = 0, loop = 3, flags = ANIMATION_PARALLEL)
+			animate(offset = f:offset - 1, time = rand() * 20 + 10)
+		if (do_after(user, 50, target = user) && user.cell.use(activationCost))
```
Comment (variableundefined) [inline code/game/objects/items/devices/chameleonproj.dm]: Spacing

**Pull request merged**